src/ZeroLog/LogEvent.cs

```diff
@@ -312,6 +458,44 @@ public void WriteToStringBufferUnformatted(StringBuffer stringBuffer)
                 stringBuffer.Append(LogManager.Config.TruncatedMessageSuffix);
         }
 
+        private void AppendJsonValue(StringBuffer stringBuffer, ref byte* dataPointer)
+        {
+            var argument = *dataPointer;
+            dataPointer += sizeof(ArgumentType);
+
+            var argumentType = (ArgumentType)(argument & ArgumentTypeMask.ArgumentType);
+
+            switch (argumentType)
+            {
+                case ArgumentType.KeyString:
+                    AppendUnformattedArgumentValue(stringBuffer, ArgumentType.String, ref dataPointer);
+                    break;
+                case ArgumentType.Guid:
+                case ArgumentType.DateTime:
+                case ArgumentType.TimeSpan:
+                    stringBuffer.Append('"');
+                    AppendUnformattedArgumentValue(stringBuffer, argumentType, ref dataPointer);
```

Comment:
This _could_ probably be OKish, but having a dedicated method for JSON would be better.

Reply:
Made this a dedicated method, thanks!